snapshots/output/syntax/src/structural-type.ts

```diff
@@ -0,0 +1,56 @@
+  export function foo(): Promise<{ member: number }> {
+// definition syntax 1.0.0 src/`structural-type.ts`/
+//documentation ```ts\nmodule "structural-type.ts"\n```
+//                ^^^ definition syntax 1.0.0 src/`structural-type.ts`/foo().
+//                documentation ```ts\nfunction foo(): Promise<{ member: number; }>\n```
+//                       ^^^^^^^ reference typescript 4.8.4 lib/`lib.es5.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.iterable.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.promise.d.ts`/Promise.
+//                       ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//                       ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2018.promise.d.ts`/Promise#
+//                                 ^^^^^^ definition syntax 1.0.0 src/`structural-type.ts`/foo().Promise:typeLiteral0:member.
+//                                 documentation ```ts\n(property) member: number\n```
+    return Promise.resolve({ member: 42 })
+//         ^^^^^^^ reference typescript 4.8.4 lib/`lib.es5.d.ts`/Promise#
+//         ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.iterable.d.ts`/Promise#
+//         ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.promise.d.ts`/Promise.
+//         ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.symbol.wellknown.d.ts`/Promise#
+//         ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2018.promise.d.ts`/Promise#
+//                 ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.promise.d.ts`/PromiseConstructor#resolve().
+//                 ^^^^^^^ reference typescript 4.8.4 lib/`lib.es2015.promise.d.ts`/PromiseConstructor#resolve().
+//                           ^^^^^^ definition syntax 1.0.0 src/`structural-type.ts`/member0:
```

Comment:
Are these two symbols deliberately different/should they be the same or have a reference relationship?

Reply:
Go to definition doesn't with tsserver for this case. I opened https://github.com/sourcegraph/scip-typescript/issues/189 because I think we can make it work anyways, and it would be cool to do better than tsserver